src/gfn/states.py

```diff
@@ -475,29 +510,381 @@ def init_forward_masks(self, set_ones: bool = True):
             self.forward_masks = torch.zeros(shape).bool()
 
 
-def stack_states(states: List[States]):
-    """Given a list of states, stacks them along a new dimension (0)."""
-    state_example = states[0]  # We assume all elems of `states` are the same.
+class GraphStates(States):
+    """
+    Base class for Graph as a state representation. The `GraphStates` object is a batched collection of
+    multiple graph objects. The `Batch` object from PyTorch Geometric is used to represent the batch of
+    graph objects as states.
+    """
+
+    s0: ClassVar[TensorDict]
+    sf: ClassVar[TensorDict]
+
+    def __init__(self, tensor: TensorDict):
+        REQUIRED_KEYS = {
+            "node_feature",
+            "edge_feature",
+            "edge_index",
+            "batch_ptr",
+            "batch_shape",
+        }
```

Comment:
Would it be better to say that the `TensorDict` should *only* contain the required keys?

Reply:
I think a user may want to implement their GraphStates, which may have additional keys.
I don't see relevant drawbacks in allowing other keys to eventually be there

Reply:
Good point!